CMakeLists.txt

```diff
@@ -23,36 +23,11 @@ project(NovelRT
 include(CheckPIESupported)
 check_pie_supported()
 
-#
-# Set the proper backend target for NovelRT
-#
-set(NOVELRT_TARGET_OPTIONS "Win32" "Linux" "macOS")
-set(NOVELRT_TARGET_OPTIONS "${NOVELRT_TARGET_OPTIONS}" CACHE INTERNAL "List of valid platforms for NovelRT")
-if(NOT DEFINED NOVELRT_TARGET)
-  message(STATUS "No backend specified - setting default based on detected OS.")
-  if(WIN32)
-    set(NOVELRT_TARGET "Win32" CACHE STRING "")
-  elseif(APPLE)
-    set(NOVELRT_TARGET "macOS" CACHE STRING "")
-    find_library(AVFOUNDATION_LIB AVFoundation)
-    find_library(FOUNDATION_LIB Foundation)
-    find_library(OBJC_LIB ObjC)
-  elseif(UNIX)
-    set(NOVELRT_TARGET "Linux" CACHE STRING "")
-  else()
-    set(NOVELRT_TARGET "Unknown" CACHE STRING "")
-  endif()
-endif()
-message("Using OS: ${NOVELRT_TARGET}")
-
-#
-# Prepend so that our FindVulkan gets picked up first when needed
-#
 list(PREPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 
-#
-# Defining options
-#
+option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
```

Comment:
Are we going back to shared libs? What happened to our reasoning for moving away from shared?